[FLINK-20426][docs] Fix broken links after Hadoop page removal (#14272)
apache/flink: docs/dev/project-configuration.zh.md

*(The only exception being when using existing Hadoop input-/output formats with Flink's Hadoop compatibility wrappers)*

If you want to use Flink with Hadoop, you need to have a Flink setup that includes the Hadoop dependencies, rather than
adding Hadoop as an application dependency. Flink will use the Hadoop dependencies specified by the `HADOOP_CLASSPATH`

> environment variable, which can usually be set by calling:

Reviewer: ```suggestion
environment variable, which can be set in the following way:
```
Minor thing: It's just that I would "call" a program but not a variable setting.
